### PR TITLE
Adds getSvpFundTransactionUnsignedHash and tests

### PR DESCRIPTION
**Pull request by jeremy-then**
Get svp fund transaction unsigned hash

### CODE REVIEW

File: rskj-core/src/test/java/co/rsk/peg/federation/FederationStorageProviderImplTests.java
Comment (julia-zack) [inline rskj-core/src/test/java/co/rsk/peg/federation/FederationStorageProviderImplTests.java:1]: there are some tests missing, following what was done [here](https://github.com/rsksmart/rskj/pull/2659/files#diff-f42b277f39730e846e15f899b88e1e7e718280d3ea3d4c4d5b035a5d66385923)

for example: `_whenThereIsNoHashSavedNorSet_shouldReturnEmpty`
`_preRSKIP419_whenProposedFederationIsSet_shouldReturnEmpty`

Comment (jeremy-then) [reply rskj-core/src/test/java/co/rsk/peg/federation/FederationStorageProviderImplTests.java:1]: Updated.

File: rskj-core/src/main/java/co/rsk/peg/federation/FederationStorageProviderImpl.java [hunk 1]
```diff
@@ -331,7 +331,28 @@ public void setLastRetiredFederationP2SHScript(Script lastRetiredFederationP2SHS
...
+
+        if (!activations.isActive(RSKIP419)) {
+            return Optional.empty();
+        }
+
+        if (svpFundTransactionUnsignedHash != null) {
+            return Optional.of(svpFundTransactionUnsignedHash);
+        }
+
+        if (isSvpFundTransactionUnsignedHashSet) {
```
Comment (marcos-iov) [inline rskj-core/src/main/java/co/rsk/peg/federation/FederationStorageProviderImpl.java:348]: I'd add a comment here explaining why it returns empty in this case

File: rskj-core/src/test/java/co/rsk/peg/federation/FederationStorageProviderImplTests.java [hunk 2]
```diff
@@ -1192,6 +1192,127 @@ void saveSvpFundTransactionUnsignedHash_postLovell700AndResettingToNull_shouldSa
...
+            Sha256Hash hash = BitcoinTestUtils.createHash(123_456_789);
+            storageAccessor.saveToRepository(SVP_FUND_TX_HASH_UNSIGNED.getKey(), BridgeSerializationUtils.serializeSha256Hash(hash));
+
+            // Act
+
+            Optional<Sha256Hash> svpFundTransactionUnsignedHash = federationStorageProvider.getSvpFundTransactionUnsignedHash(arrowheadActivations);
+
+            // Assert
+
+            assertEquals(Optional.empty(), svpFundTransactionUnsignedHash);
```
Comment (marcos-iov) [inline rskj-core/src/test/java/co/rsk/peg/federation/FederationStorageProviderImplTests.java:1227]: Let's try to keep a standard. The test below uses `assertFalse(svpFundTransactionUnsignedHash.isPresent());`

File: rskj-core/src/test/java/co/rsk/peg/federation/FederationStorageProviderImplTests.java [hunk 3]
```diff
@@ -1192,6 +1192,127 @@ void saveSvpFundTransactionUnsignedHash_postLovell700AndResettingToNull_shouldSa
...
+        @Test
+        void getSvpFundTransactionUnsignedHash_postLovell700AndSetToNull_shouldReturnEmpty() {
+
+            // Arrange
+
+            federationStorageProvider.setSvpFundTransactionUnsignedHash(null);
+
+            // Act
+
+            Optional<Sha256Hash> svpFundTransactionUnsignedHash = federationStorageProvider.getSvpFundTransactionUnsignedHash(activations);
```
Comment (marcos-iov) [inline rskj-core/src/test/java/co/rsk/peg/federation/FederationStorageProviderImplTests.java:1306]: I think we should also test the case when a `null` value is in storage

File: rskj-core/src/main/java/co/rsk/peg/federation/FederationStorageProviderImpl.java (hunk 1 above)
Comment (julia-zack) [reply rskj-core/src/main/java/co/rsk/peg/federation/FederationStorageProviderImpl.java:348]: agree

File: rskj-core/src/test/java/co/rsk/peg/federation/FederationStorageProviderImplTests.java [hunk 4]
```diff
@@ -1192,6 +1192,127 @@ void saveSvpFundTransactionUnsignedHash_postLovell700AndResettingToNull_shouldSa
...
+            Optional<Sha256Hash> svpFundTransactionUnsignedHash = federationStorageProvider.getSvpFundTransactionUnsignedHash(arrowheadActivations);
+
+            // Assert
+
+            assertEquals(Optional.empty(), svpFundTransactionUnsignedHash);
+
+        }
+
+        @Test
+        void getSvpFundTransactionUnsignedHash_whenThereIsNoProposedFederationSavedNorSet_shouldReturnEmpty() {
```
Comment (julia-zack) [inline rskj-core/src/test/java/co/rsk/peg/federation/FederationStorageProviderImplTests.java:1232]: you left the proposedFederation name jaja

File: rskj-core/src/main/java/co/rsk/peg/federation/FederationStorageProvider.java [hunk 5]
```diff
@@ -44,6 +44,8 @@ public interface FederationStorageProvider {
 
     void setSvpFundTransactionUnsignedHash(Sha256Hash hash);
 
+    Optional<Sha256Hash> getSvpFundTransactionUnsignedHash(ActivationConfig.ForBlock activations);
```
[the remark this answers is not in the document]
Comment (jeremy-then) [reply rskj-core/src/main/java/co/rsk/peg/federation/FederationStorageProvider.java]: Updated.

File: rskj-core/src/main/java/co/rsk/peg/federation/FederationStorageProviderImpl.java (hunk 1 above)
Comment (jeremy-then) [reply rskj-core/src/main/java/co/rsk/peg/federation/FederationStorageProviderImpl.java]: Updated.

File: rskj-core/src/test/java/co/rsk/peg/federation/FederationStorageProviderImplTests.java (hunk 2 above)
Comment (jeremy-then) [reply rskj-core/src/test/java/co/rsk/peg/federation/FederationStorageProviderImplTests.java]: Updated.

File: rskj-core/src/test/java/co/rsk/peg/federation/FederationStorageProviderImplTests.java (hunk 3 above)
Comment (jeremy-then) [reply rskj-core/src/test/java/co/rsk/peg/federation/FederationStorageProviderImplTests.java]: Updated.

File: rskj-core/src/test/java/co/rsk/peg/federation/FederationStorageProviderImplTests.java (hunk 4 above)
Comment (jeremy-then) [reply rskj-core/src/test/java/co/rsk/peg/federation/FederationStorageProviderImplTests.java]: Updated.

**Review by marcos-iov — APPROVED** [review]

**Review by nathanieliov — APPROVED** [review]
Left a couple of suggestions and a fix for a typo.

**Review by marcos-iov — APPROVED** [review]
